src/Core/Domain/Product/Stock/ValueObject/DeltaQuantity.php

```diff
@@ -0,0 +1,88 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Domain\Product\Stock\ValueObject;
+
+use PrestaShop\PrestaShop\Core\Domain\Product\Stock\Exception\ProductStockConstraintException;
+
+/**
+ * Holds delta quantity for a product update along with a stock movement reason
+ */
+class DeltaQuantity
+{
+    /**
+     * @var int
+     */
+    private $deltaQuantity;
+
+    /**
+     * @var MovementReasonId
+     */
+    private $movementReasonId;
+
+    /**
+     * @param int $deltaQuantity
+     * @param MovementReasonId $movementReasonId
+     */
+    public function __construct(
+        int $deltaQuantity,
+        MovementReasonId $movementReasonId
+    ) {
+        $this->assertValueIsNotZero($deltaQuantity);
+        $this->deltaQuantity = $deltaQuantity;
+        $this->movementReasonId = $movementReasonId;
+    }
+
+    /**
+     * @return int
+     */
+    public function getDeltaQuantity(): int
```

Comment:
```suggestion
    public function getValue(): int
```

Reply:
It is a bit "cosmetic" but I'd like to avoid the double methods like `$properties->getDeltaQuantity()->getDeltaQuantity()` It's totally subjective but it makes my eyes bleed 😅

Reply:
On second thought since this VO is a combination of delta quantity and movement reason it's probably the class itself which should be renamed, because it's more than a delta quantity It's more a "stock movement"

Reply:
So I would suggest renaming this VO `StockModification`, along with its associated getters/setters This way it reflects more precisely its purpose, and we would call `$properties->getStockModification()->getDeltaQuantity()`

Reply:
you really don't want me to merge anything this year, don't you :smile: